python/python_direct/README.md

```diff
@@ -0,0 +1,44 @@
+<!--
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+-->
+# Direct Bindings API
+
+The Direct Python API maps from CPP to Python directly, so any CPP function can be exposed to python users.
+
+### Legacy Python API
+* A Trie data structure is used to map a fusion math definition to FusionExecutorCache.
+* This FusionCache creates the CPP Fusion. The user cannot directly create nor modify the CPP fusion in python.
+* RecordFunctor is created for each operation to handle equivalence, hashing, printing, and CPP interoperability.
+
+### Differences
+* Caching fusions by definitions AND serialization is not supported with direct bindings.
+* Supports using constant scalars with any operation.
+* The python reproducers map directly to Fusion IR, so it will be more verbose. Direct bindings does not have `RecordFunctor` objects to map from Fusion IR to original python API.
+
+## How to add Fusion IR?
+* Create `py::class_` for IR node.
+* `Statement`, `Expr`, `Val`, `IterDomain`, `TensorDomain`, `TensorView`, and `Scalar` exist in `python_direct/ir.cpp`
+* All other nodes from `csrc/ir/internal_nodes.h` go to `python_direct/internal_ir.cpp`
+* Add necessary IR node operations to `pyt::class_`
```

Comment:
I think it's worth noting that: This is really only necessary when we need expose the node in python. Which isn't necessary unless we try to schedule it.